lib/matplotlib/textpath.py

```diff
@@ -243,25 +242,28 @@ def get_glyphs_tex(self, prop, s, glyph_map=None,
 
         # Gather font information and do some setup for combining
         # characters into strings.
-        for x1, y1, dvifont, glyph, width in page.text:
-            font, enc = self._get_ps_font_and_encoding(dvifont.texname)
-            char_id = self._get_char_id(font, glyph)
-
+        for text in page.text:
+            font = get_font(text.font_path)
+            char_id = self._get_char_id(font, text.glyph)
             if char_id not in glyph_map:
+                spec = text.glyph_spec
                 font.clear()
                 font.set_size(self.FONT_SCALE, self.DPI)
-                # See comments in _get_ps_font_and_encoding.
-                if enc is not None:
-                    index = font.get_name_index(enc[glyph])
+                # See comments in Text.get_glyph_spec.
+                if isinstance(spec, str):
+                    index = font.get_name_index(spec)
                     font.load_glyph(index, flags=LOAD_TARGET_LIGHT)
-                else:
-                    font.load_char(glyph, flags=LOAD_TARGET_LIGHT)
+                elif isinstance(spec, int):
+                    self._select_native_charmap(font)
+                    font.load_char(spec, flags=LOAD_TARGET_LIGHT)
+                else:  # Should not occur.
+                    raise TypeError("Unexpected glyph spec type")
```

Comment:
Would it make sense to print some info about the spec here? In the unlikely event that it happens to someone...

Reply:
sure